src/libponyc/expr/control.c

```diff
@@ -367,8 +367,11 @@ bool expr_recover(ast_t* ast)
 
   if(r_type == NULL)
   {
-    ast_error(ast, "can't recover to this capability");
-    ast_error(expr, "expression type is %s", ast_print_type(type));
+    errorframe_t frame = NULL;
+    ast_error_frame(&frame, ast, "can't recover to this capability");
+    ast_error_frame(&frame, expr, "expression type is %s",
+      ast_print_type(type));
+    errorframe_report(&frame);
```

Comment:
Incidentally, I also fixed the way this error gets reported.  It was previously being reported as two separate errors, but it will now be reported as an `Error: ` message followed by an additional indented `Info: ` message, which is as it should be.